main.py

```diff
@@ -144,65 +155,79 @@ async def transform_stream_for_client(
                         continue
 
                     url_set.add(doc.metadata["source"])
-                    yield doc.metadata["title"] + ":" + doc.metadata["source"] + "\n"
-
-                if len(url_set) > 0:
-                    yield "SOURCES:----------------------------"
+                    all_sources.append(
+                        doc.metadata["title"] + ":" + doc.metadata["source"]
+                    )
+                if all_sources:
+                    src = {"sources": "\n".join(all_sources)}
```

Comment:
Figured it's easier to use an established format than use our own string parsing?